io/include/pcl/io/pcd_io.h

```diff
@@ -252,12 +252,14 @@ namespace pcl
                             const Eigen::Quaternionf &orientation);
 
       /** \brief Generate the header of a BINARY_COMPRESSED PCD file format
+        * \param[out] os the stream into which to write the header
         * \param[in] cloud the point cloud data message
         * \param[in] origin the sensor acquisition origin
         * \param[in] orientation the sensor acquisition orientation
         */
-      std::string
-      generateHeaderBinaryCompressed (const pcl::PCLPointCloud2 &cloud,
+      int
```

Comment:
My primary concerns were: consistency with the rest of the code in that file, and limiting the scope of my changes (which consist mostly of shuffling the existing code).  Much of this code is not how I'd have written it.  I'm happy to change anything you'd like (though I hope we can limit the scope, for this round).

So you want it to return void, and simply document that the state of the stream should be checked to determine success/failure?